diff -ur temp/vanilla/binutils-2.9.1/bfd/aix386-core.c ./binutils-2.9.1/bfd/aix386-core.c
--- temp/vanilla/binutils-2.9.1/bfd/aix386-core.c	Fri May  1 08:47:59 1998
+++ ./binutils-2.9.1/bfd/aix386-core.c	Fri May 25 10:26:44 2018
@@ -250,7 +250,7 @@
     "aix386-core",
     bfd_target_unknown_flavour,
     BFD_ENDIAN_BIG,		/* target byte order */
-    BFD_ENDIANG_BIG,		/* target headers byte order */
+    BFD_ENDIAN_BIG,		/* target headers byte order */
   (HAS_RELOC | EXEC_P |		/* object flags */
    HAS_LINENO | HAS_DEBUG |
    HAS_SYMS | HAS_LOCALS | WP_TEXT),
