diff -ur --no-dereference orig/include/stddef.h mod/include/stddef.h
--- orig/include/stddef.h	2023-01-10 19:07:10.232118326 -0800
+++ mod/include/stddef.h	2023-01-10 19:10:05.233202593 -0800
@@ -163,7 +163,10 @@
 #define __SIZE_TYPE__ long unsigned int
 #endif
 #if !(defined (__GNUG__) && defined (size_t))
+#ifndef _IS_SIZE_T_
+#define _IS_SIZE_T_
 typedef __SIZE_TYPE__ size_t;
+#endif
 #endif /* !(defined (__GNUG__) && defined (size_t)) */
 #endif /* __size_t */
 #endif /* _SIZET_ */
@@ -238,10 +241,13 @@
 #define __WCHAR_TYPE__ int
 #endif
 #ifndef __cplusplus
+#ifndef _IS_WCHAR_T_
+#define _IS_WCHAR_T_
 typedef __WCHAR_TYPE__ wchar_t;
 #endif
 #endif
 #endif
+#endif
 #endif
 #endif
 #endif
diff -ur --no-dereference orig/include/stdlib.h mod/include/stdlib.h
--- orig/include/stdlib.h	2023-01-10 19:07:10.232118326 -0800
+++ mod/include/stdlib.h	2023-01-10 18:52:57.000000000 -0800
@@ -58,9 +58,15 @@
 #define __WCHAR_TYPE__ int
 #endif
 #ifndef __cplusplus
+#ifndef _IS_WCHAR_T_
+#define _IS_WCHAR_T_
 typedef __WCHAR_TYPE__ wchar_t;		/* As per ANSI */
 #endif
+#endif
+#ifndef _IS_MBCHAR_T_
+#define _IS_MBCHAR_T_
 typedef unsigned long mbchar_t;
+#endif
 #ifndef __SIZE_TYPE__
 #define __SIZE_TYPE__ long unsigned int
 #endif
@@ -78,7 +84,13 @@
 #endif /* MB_CUR_MAX */
 
 extern int mblen(const char *, size_t);
+#ifndef _IS_MBTOWC_
+#define _IS_MBTOWC_
 extern int mbtowc(wchar_t *, const char *, size_t);
+#endif
 extern int wctomb(char *, wchar_t);
+#ifndef _IS_MBSTOWCS_
+#define _IS_MBSTOWCS_
 extern size_t mbstowcs(wchar_t *, const char *, size_t);
+#endif
 extern size_t wcstombs(char *, const wchar_t *, size_t);
