diff -ur --no-dereference orig/usr/include/ctab.h mod/usr/include/ctab.h
--- orig/usr/include/ctab.h	1990-02-21 14:50:22.000000000 -0800
+++ mod/usr/include/ctab.h	2023-01-07 14:50:20.410057970 -0800
@@ -17,9 +17,17 @@
 #ifndef _H_CTAB
 #define _H_CTAB
 #ifndef _h_NLCHAR
-typedef unsigned char UCHAR;	/* redefined to avoid including NLchar.h */
+
+#ifndef _IS_UCHAR_
+#define _IS_UCHAR_
+typedef unsigned char UCHAR;    /* redefined to avoid including NLchar.h */
+#endif
+
 #ifdef _MBCS
-typedef unsigned long wchar_t;	/* redefined to avoid including mbcs.h */
+#ifndef _IS_WCHAR_T_
+#define _IS_WCHAR_T_
+typedef int wchar_t;     /* redefined to avoid including mbcs.h */
+#endif
 struct NLctab {
 	long	ct_maxctp;		/* Number of chars for ctype */
 	long	ct_maxcol;		/* Number of chars for collation */
diff -ur --no-dereference orig/usr/include/mbcs.h mod/usr/include/mbcs.h
--- orig/usr/include/mbcs.h	1990-02-21 14:50:24.000000000 -0800
+++ mod/usr/include/mbcs.h	2023-01-07 14:50:20.410057970 -0800
@@ -56,7 +56,10 @@
 /*
  * Typedefs specific to mbcs.
  */
+#ifndef _IS_MBCHAR_T_
+#define _IS_MBCHAR_T_
 typedef unsigned long mbchar_t;
+#endif
 
 /*
  * This is a per character conversion entry structure.  It is used in the
@@ -252,8 +255,14 @@
 extern char	*mbspbrk(const char *, const char *);
 extern char	*mbstok(char *, const char *);
 
+#ifndef _IS_MBTOWC_
+#define _IS_MBTOWC_
 extern int	mbtowc(wchar_t *, const char *, size_t);
+#endif
+#ifndef _IS_MBSTOWCS_
+#define _IS_MBSTOWCS_
 extern size_t	mbstowcs(wchar_t *, const char *, size_t);
+#endif
 extern size_t	mbslen(const char *);
 extern int	mblen(const char *, size_t);
 extern int	mbdwidth(const char *);
diff -ur --no-dereference orig/usr/include/stddef.h mod/usr/include/stddef.h
--- orig/usr/include/stddef.h	1990-02-21 14:50:26.000000000 -0800
+++ mod/usr/include/stddef.h	2023-01-07 14:50:20.413391382 -0800
@@ -10,14 +10,25 @@
  */
 /* stddef.h */
 
-typedef int ptrdiff_t;
-typedef long size_t;
-
-#define NULL	0
+#ifndef _STDDEF_H_
+#define _STDDEF_H_
 
-#define offsetof(struct_s, member_m) (size_t)(&((struct_s *)0)->member_m)
+typedef int ptrdiff_t;
 
+#ifndef _IS_SIZE_T_
+#define _IS_SIZE_T_
+#ifndef __SIZE_TYPE__
+#define __SIZE_TYPE__ long
+#endif
+typedef __SIZE_TYPE__ size_t;
+#endif
 /*
  * MBCS typedefs 
  */
 typedef unsigned long	wchar_t;	/* wide character */
+
+#define NULL	0
+
+#define offsetof(struct_s, member_m) (size_t)(&((struct_s *)0)->member_m)
+
+#endif
diff -ur --no-dereference orig/usr/include/stdio.h mod/usr/include/stdio.h
--- orig/usr/include/stdio.h	1990-02-21 14:50:26.000000000 -0800
+++ mod/usr/include/stdio.h	2023-01-07 14:50:20.413391382 -0800
@@ -138,7 +138,9 @@
 #define	TMP_MAX	4096
 #define	FOPEN_MAX 200
 
+#ifndef __SIZE_TYPE__
 typedef	long	size_t;
+#endif
 
 #ifdef CC4
 #define _bufend(p)	_bufendtab[(p)->_file]
@@ -224,9 +226,11 @@
 #endif
 #endif
 
+#ifndef __WCHAR_TYPE__
 typedef unsigned long wchar_t;			/* This is against ansi rule */
 						/* except we need wchar_t    */
 						/* defined for out functions */
+#endif
 
 #ifndef	lint					/* proto-types do not work   */
 						/* with lint                 */
diff -ur --no-dereference orig/usr/include/string.h mod/usr/include/string.h
--- orig/usr/include/string.h	1990-02-21 14:50:26.000000000 -0800
+++ mod/usr/include/string.h	2023-01-07 14:50:20.433391848 -0800
@@ -12,7 +12,13 @@
  */
 
 /*  POSIX needed definition of size_t */
-typedef long	size_t;
+#ifndef _IS_SIZE_T_
+#define _IS_SIZE_T_
+#ifndef __SIZE_TYPE__
+#define __SIZE_TYPE__ long
+#endif
+typedef __SIZE_TYPE__	size_t;
+#endif
 
 extern char
 	*strcpy(),
diff -ur --no-dereference orig/usr/include/sys/NLchar.h mod/usr/include/sys/NLchar.h
--- orig/usr/include/sys/NLchar.h	1990-02-21 14:50:26.000000000 -0800
+++ mod/usr/include/sys/NLchar.h	2023-01-07 14:50:20.433391848 -0800
@@ -28,8 +28,15 @@
 /*  The big type itself.
  */
 
+#ifndef _IS_NLCHAR_
+#define _IS_NLCHAR_
 typedef wchar_t NLchar;
+#endif
+
+#ifndef _IS_UCHAR_
+#define _IS_UCHAR_
 typedef unsigned char  UCHAR;
+#endif
 
 /*  Number of distinct NLchars.
  */
diff -ur --no-dereference orig/usr/include/sys/types.h mod/usr/include/sys/types.h
--- orig/usr/include/sys/types.h	1990-02-21 14:50:41.000000000 -0800
+++ mod/usr/include/sys/types.h	2023-01-07 14:50:20.433391848 -0800
@@ -71,7 +71,13 @@
 /* Miscellaneous kernel types */
 typedef	long	  daddr_t;	/* Disk address */
 typedef	char *	  caddr_t;	/* Kernel virtual address (char *) */
-typedef	long	  size_t;	/* Number of mem allocation units (clicks) */
+#ifndef _IS_SIZE_T_
+#define _IS_SIZE_T_
+#ifndef __SIZE_TYPE__
+#define __SIZE_TYPE__ long
+#endif
+typedef	__SIZE_TYPE__  size_t;	/* Number of mem allocation units (clicks) */
+#endif
 typedef	size_t	  click_t;	/* Number of mem allocation units (clicks) */
 typedef	daddr_t	  swblk_t;	/* Swap block */
 typedef	long	  time_t;	/* Time in seconds (since Jan. 1, 1970) */
